### DISCUSSION TITLE
Invoke-PnPSiteTemplate Handlers for apps

### DISCUSSION THREAD

**Discussion by ZARSYNTEX** [pnp/powershell#3117] [General]
Hi all,
in which handler of Invoke-PnPSiteTemplate are SharePoint apps included?

Currently I use following command and apps are missing.

Invoke-PnPSiteTemplate -Path $templatefile -ClearNavigation:$true -OverwriteSystemPropertyBagValues:$true -Handlers WebSettings, SupportedUILanguages, SiteSettings, SiteSecurity, SiteHeader, SiteFooter, SearchSettings, RegionalSettings, PropertyBagEntries, Pages, PageContents, Navigation, Lists, Files, Fields, ContentTypes

Best regards
Robin

**Comment by AndersRask** [comment]
I can see Get-PnPSiteTemplate extracts apps in 
```

<pnp:ApplicationLifecycleManagement>
        <pnp:Apps>
          <pnp:App
```

but cannot see a specific handler for it.
Did you try to use `Invoke-PnPSiteTemplate` with all handlers using a "raw" export that you know has app instances?

If they do not include them, then they are probably not supported.
You will then have to install them after finishing the site provisioning using` Install-PnPApp`.
Remember to publish them to the app catalog if it is another teant.
